src/cron-job.ts

```diff
@@ -0,0 +1,251 @@
+
+  /**
+   * Specifies the number of failed jobs history retained.
+   *
+   * @default - 1.
+   */
+  readonly failedJobsRetained?: number;
+}
+
+/**
+ * A CronJob is responsible for creating a Job and scheduling it based on provided scheduled. This helps running Jobs in a recurring manner.
+ */
+export class CronJob extends workload.Workload {
+  /**
+   * Properties of the recurring `Job`.
+   */
+  public readonly jobProperties: JobProps;
+
+  /**
+   * Schedule for which to run the recurring job. By default, the job would be scheduled to run every minute.
+   */
+  public readonly schedule?: CronOptions;
+
+  /**
+   * Timezone in which the recurring job is scheduled.
+   */
+  public readonly timeZone?: string;
+
+  /**
+   * Concurrency policy for the recurring job. This specifies how the job scheduling would be handled based on which policy is selected.
+   */
+  public readonly concurrencyPolicy?: string;
+
+  /**
+   * Deadline to start next instance of job after the expected scheduling time. The job is considered as failed if it misses this deadline.
+   */
+  public readonly startingDeadline?: Duration;
+
+  /**
+   * Flag that tells to suspend the upcoming executions of the job.
+   */
+  public readonly suspend?: boolean;
+
+  /**
+   * The number of successful jobs to retain.
+   */
+  public readonly successfulJobsRetained?: number;
+
+  /**
+   * The number of failed jobs to retain.
+   */
+  public readonly failedJobsRetained?: number;
+
+  /**
+   * @see base.Resource.apiObject
+   */
+  protected readonly apiObject: ApiObject;
+
+  /**
+   * Represents the resource type.
+   */
+  public readonly resourceType = 'cronjob';
+
+  constructor(scope: Construct, id: string, props: CronJobProps) {
+    super(scope, id, {
+      restartPolicy: RestartPolicy.ON_FAILURE,
+      select: false, //Why is this false in Job?
+      ...props,
+    });
+
+    this.apiObject = new k8s.KubeCronJob(this, 'Resource', {
+      metadata: props.metadata,
+      spec: Lazy.any({ produce: () => this._toKube() }),
+    });
+
+    if (!this.isValidTimezone(props.timeZone)) {
+      throw new Error(`Invalid timezone: ${props.timeZone}`);
+    }
+
+    this.jobProperties = props.jobProperties;
+    this.schedule = props.schedule;
+    this.timeZone = props.timeZone;
+    this.concurrencyPolicy = props.concurrencyPolicy;
+    this.startingDeadline = props.startingDeadline;
+    this.suspend = props.suspend;
+    this.successfulJobsRetained = props.successfulJobsRetained;
+    this.failedJobsRetained = props.failedJobsRetained;
+  }
+
+  /**
+   * @internal
+   */
+  public _toKube(): k8s.CronJobSpec {
+    return {
+      concurrencyPolicy: this.concurrencyPolicy ?? ConcurrencyPolicy.FORBID,
+      failedJobsHistoryLimit: this.failedJobsRetained,
+      jobTemplate: {
+        metadata: this.jobProperties.metadata,
+        spec: this.getJobSpec(),
+      },
+      schedule: this.getCronExpression(this.schedule),
```

Comment:
The `getCronExpression` method shouldn't exist. This should just be:

```suggestion
      schedule: this.schedule.toCronExpression(),
```

And we should add `toCronExpression` on the [`Schedule`](https://github.com/cdk8s-team/cdk8s-core/blob/2.x/src/schedule.ts) class. Also looking at that class, I think we got it wrong because it provide functionality not available in Kubernetes (like rates). We need to first fix it.

Reply:
Yes, on it.